### ISSUE TITLE
Tier 4 Plastic Idea: Kevlar Composite

### ISSUE THREAD

**Issue by Spluff5** [GTNewHorizons/GT-New-Horizons-Modpack#10016]
### Your GTNH Discord Username

Spluff#2951

### Your Pack Version

2.1.2.0

### Your Proposal

In the interest of adding more content to the late game, I'm proposing a new plastic to replace polybenzimidazole in UHV and UEV machines. This recipe is designed to be the ultimate in oil-based chemistry, being roughly twice the size of the PBI recipe line.

### Your Goal

Currently in the pack, every two voltage tiers the type of plastic needed to make things such as machine hulls changes, with polyethylene for HV and EV, and Polytetrafluoroethylene for IV and LuV. However, polybenzimidazole is used for the next six tiers after this, meaning new plastic doesn't need to be developed after the player reaches ZPM.

The reason I'm proposing this particular recipe is because Kevlar, in real life, is one of the strongest plastics known. This recipe chain also gives use to a few materials that are already in the pack but have very little use, such as bromine, aniline, nitrobenzene, propanoic acid, formaldehyde, and butanol.

### Your Vision

# Tier 4 Plastic Idea: Kevlar Composite
![Tier 4 Plastic (Kevlar)](https://user-images.githubusercontent.com/46005083/159638405-cf26a054-0380-4df2-96af-7aa36100ec04.png)
Below are a series of major precursors that go into the recipe chain. Most of these are new items or fluid that would need to be added to the game. Some items are only catalysts and so only a little bit may be used in the recipe. Normally with other recipes already in the pack, tiny dusts are used for item catalysts, and 10x less fluid compared to other ingredients is used for fluid catalysts. Some recipes in the flowchart are listed as Chemical Reactor (CR), while others are Large Chemical Reactor (LCR). Recipes with more than 3 ingredients need the LCR, but it’s up to you which of the other recipes should require the LCR.

## Liquid Crystal Kevlar
New light yellow fluid. Could have a unique fluid texture because it’s liquid crystal. Created by reacting 1,4-Phenylenediamine and Terephthaloyl Chloride with N-Methylpyrrolidone and Calcium Chloride as catalysts.

### 1,4-Phenylenediamine
This is one of the two chemicals that forms the actual Kevlar polymer.
#### Chlorobenzene
- Chlorobenzene is obtained through the same method as it is for the polybenzimidazole line, either being pumped from mars or created in the chemical reactor from Chlorine and Benzene.
#### Sodamide
- Sodamide is a very strong alkaline that can be used to attach amino groups to benzene rings. It's a new pale-blue solid. 
- Iron(III) Nitrate is needed as a catalyst to form the sodamide. It's a new pale orange liquid
#### 1,4-Phenylenediamine
1. 4-Nitrochlorobenzene (new lime green fluid) is made by reacting Nitration Mixture with Benzene. This is the same recipe as 2-Nitrochlorobenzene used in the PBI line, but with a different programmed circuit
2. 4-Nitroaniline (new grey-blue fluid) is made by reacting 4-Nitrochlorobenzene with Sodamide Dust. This has a salt byproduct
3. 1,4-Phenylenediamine (new grey-blue fluid) is made by reacting 4-Nitroaniline with Hydrogen. Palladium dust is used here as a catalyst (using small or tiny dusts may be more appropriate)

### Terephthaloyl Chloride
This is one of the two chemicals that forms the actual Kevlar polymer.
#### Bromine
Bromine might be obtained in one of two ways depending on which we think is better:
- Electrolyzing 20,000 mB of Brine in the Multiblock Electrolyzer for 30 mB of Bromine Fluid
- Electrolyzing Bromine-containing ore dusts. GT++ adds four of these ores to the game but they don't currently spawn as part of any vein. A new ore vein would have to be added to the game, probably on Io. Assuming GT++ ores can't be added to a new vein, new ores would have to be added. My recommendation is that this is a MercuryVein, with Cinnabar Ore as Primary, Grechishchevite (new ore, orange-red) as Secondary,  Dolomite as Between, and Pyrite as Sporadic. In the future this could change.
#### Dimethylbenzene
- Simply produced in the same way it is for the PBI line. No new recipes required.
#### Thionyl Chloride
- Thionyl Chloride (new pale-yellow fluid) is a chemical that adds chlorine atoms to organic molecules. It is made by reacting Sulfur Trioxide, Chlorine, and Sulfur Dichloride.
- Sulfur Dichloride (new dark-red fluid) is made by reacting chlorine with sulfur
#### Terephthaloyl Chloride
1. Bromine reacts with Hydrogen to give Hydrobromic Acid (new red fluid)
2. Benzenedicarboxylic Acid (new dark-green fluid) is made by reacting Oxygen and Dimethylbenzene. Cobalt Dust, Manganese Dust, and Hydrobromic Acid are all included as catalysts.
3. Terephthalic Acid (new green-black fluid) is obtained from Benzenedicarboxylic Acid in the distillation tower. This has a byproduct of Phthalic Acid, which is already used in the PBI line.
4. Dimethyl Terephthalate (new green-grey fluid) is obtained from reacting Terephthalic Acid with Sulfuric Acid and Methanol.
5. Terephthaloyl Chloride (new bright-green fluid) is obtained by reacting Dimethyl Terephthalate with Thionyl Chloride

### N-Methylpyrrolidone
This is a catalyst needed to make the Kevlar form from Terephthaloyl Chloride and 1,4-Phenylenediamine
#### Methylamine
- Methylamine (new blue fluid) is made by reacting Ammonia with Methanol. Alumina Silicate Wool is needed as a catalyst
#### N-Methylpyrrolidone
1. Acetaldehyde (new grey fluid) is made by reacting Oxygen with Ethylene on Circuit 8. The circuit is important as other numbers create polyethylene and acetic acid
2. beta-Butyrolactone (new reddish-purple fluid) is made by reacting Acetaldehyde with Ethenone
3. N-Methylpyrrolidone (new purple fluid) is made by reacting Methylamine with beta-Butyrolactone

## Polyurethane Resin
Created by reacting 4,4'-Methylene Diphenyldiisocyanate with Ethylene Glycol and Pentaerythritol. Dibutyltin Catalyst is included as a catalyst. This fluid already exists in the game (added by GT++), but it is called “polyurethane” instead of “polyurethane resin” and has no recipe currently.

### 4,4'-Methylene Diphenyldiisocyanate
This is the main building block of the resin
#### Phosgene
- Phosgene (new pale-green gas) is created by reacting carbon monoxide with Chlorine
#### Aniline
Both Aniline and Nitrobenzene already exist in the game (added by GT++), but currently don’t have recipes.
- Aniline could be created by reacting Nitrobenzene with Hydrogen using a Palladium Dust catalyst.
- Nitrobenzene could be created by reacting Benzene with Nitration Mixture
#### 4,4'-Methylene Diphenyldiisocyanate
1. Methylene Diphenyldiisocyanate Mixture (new light-purple fluid) is created by reacting Hydrochloric Acid and Formaldehyde with Aniline and Phosgene
2. 4,4'-Methylene Diphenyldiisocyanate (new purple fluid) is created in the distillery from Methylene Diphenyldiisocyanate Mixture. Only a little bit of 4,4'-Methylene Diphenyldiisocyanate is created, although the exact amount can be balanced.

### Dibutyltin Catalyst
This catalyst is needed for the formation of the resin. Both butanol and Propanoic Acid currently exist in the game (Added by GT++) but have no recipes
#### Butanol
- Butanol is formed by reacting Butene with Steam. An alternative recipe could react Butene or Butane with Oxygen using a specific programmed circuit but I chose this recipe because it mirrors the recipe for Ethanol
#### Propanoic Acid
- Propanoic Acid is formed by reacting Oxygen with Propene using circuit 9
#### Dibutyltin Catalyst 
1. Dibutyltin Catalyst (new white fluid) is formed by reacting Butanol with Propanoic Acid and Tin Dust, using Iron III Chloride as a catalyst.

### Ethylene Glycol
This is a secondary building block of the resin
1. Ethylene is reacted with Oxygen and Silver Dust catalyst using circuit 6 to give Ethylene Oxide (new grey fluid)
2. Ethylene Oxide reacts with Water on circuit 1 to give Ethylene Glycol (new light grey fluid)

### Silicone Oil
Silicone Oil is a surfactant that helps the polyurethane resin ingredients mix together and react better. It is used like a catalyst. This fluid could also be used to replace Lubricant in high-tier recipes.
1. Silicone Oil (new grey fluid) is made by reacting Water, Ethylene Oxide, and Dimethylchlorosilane on circuit 24

### Pentaerythritol
Pentaerythritol is an additive to the resin to make it more rigid
1. Pentaerythritol (new pink fluid) is formed by reacting Formaldehyde with Sodium Hydroxide and Acetaldehyde

## Kevlar Plate
Kevlar plates are formed similarly to fiber-reinforced epoxy using the chemical bath. From the plate, other components can be made.

1. Liquid Crystal Kevlar is first made into fibres in the fluid solidifier. These fibres are effectively the “fine wire” of Kevlar, but can’t be melted and cast to make the plates without using polyurethane resin. Instead of a mould in the solidifier, a new item called the Spinneret Module is used. This item pumps the fluid at high pressure through tiny holes to make the fibres. It can be made in the Assembler using Pumps, Wire Extruder Shape, Osmiridium Rings, Polybenzimidazole Tiny Pipes, and other items we think are appropriate.
2. Kevlar Fibers are made into sheets of Woven Kevlar fabric in the Assembler. These could replace Wool or other fabrics in high tier recipes
3. Kevlar Plates are made by combining Woven Kevlar and Polyurethane Resin in the Chemical Bath

Polybenzimidazole will tremble before Kevlar!

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this feature please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I believe there is nothing similar in the pack already, or the existing solution isn't good enough.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members thinking it's inappropriate.
- [x] I believe this feature would make the pack better.

**Comment by chochem** [comment]
This did not turn out very well in my opinion. While the chemistry might be interesting, the gameplay is not. This is just spam of very basic machines with nothing interesting going on. No new multiblock, no new mechanic, no intricate redstone control required, no interaction with TecTech or bartworks or magic mods or planets.  Really from a gameplay perspective there is nothing here that engages the player.

This is just tedious to build but not interesting or difficult. And once built it adds a lot of TPS demand.

**Comment by Dream-Master** [comment]
> This did not turn out very well in my opinion. While the chemistry might be interesting, the gameplay is not. This is just spam of very basic machines with nothing interesting going on. No new multiblock, no new mechanic, no intricate redstone control required, no interaction with TecTech or bartworks or magic mods or planets. Really from a gameplay perspective there is nothing here that engages the player.
> 
> This is just tedious to build but not interesting or difficult. And once built it adds a lot of TPS demand.

My coding knowlege is so litte that i cant add super cool and fancy new machines and stuff. If someone is better in coding feel free to add recipes to new machines. I just add the recipes and sute spawning more lcr recipes are boring.

**Comment by chochem** [comment]
Yes, I get that. My coding knowledge is not great either and I am not claiming to have a solution at hand anyway.

But I do think kevlar still needs something a bit more interesting than just pure length. An option that is _comparatively_ easy to implement would be to use various more complex machines from mods where appropriate (dissolution tank, digester, chem plant, bacterial vat?) and in turn reduce the number of steps a bit. But really I am hoping some other people have even better ideas.

From what I understand kevlar will not be required for anything yet in the upcoming stable update, so there is still time for suggestions and improvements. I don't think many people have really looked at the processing line yet.

**Comment by Dream-Master** [comment]
yes i can add stuff into existing machines as well. reduce the steps are not the goal. Make kevlar need alot of steps but with more interesting machines indeed

**Comment by POPlol333** [comment]
To help a bit with boring casual chemical reactor spaming we could try reworking a bit LCR.
As now LCR Structure check accept all the coil including the latest colen added, so I've been thinking, why not use them as a recipe gate like we do in EBF? this time not as heat gate but as "something-ish I honestly don't know", could be lore explained like "the precision engine of the chemical reactor need better coil to cool off so he don't fail his chemical reaction.
So basically this suggestion could apply largely to all the LCR recipe and a bit more unique than just a multi chemical reactor

**Comment by POPlol333** [comment]
Or maybe use that idea to make an "advanced chemical reactor" machine

**Comment by POPlol333** [comment]
Some details, as LCR are already kind of overpowered, for now coils are purely aesthetics and beyond that recipe gate I don't want them to add even more overclock between recipes

**Comment by BritishCynic** [comment]
As this relates to the original suggestion, this can now be closed. I'd suggest that if anyone has further issues related to improving the Kevlar line that a new issue is created for them.
